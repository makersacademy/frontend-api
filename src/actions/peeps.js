import store from '../store/configureStore'

export const addPeep = (peep) => ({
  type: 'ADD_PEEP',
  peep
})

export const startAddPeep = (peepData = {}) => {
  return (dispatch) => {
    const session = store.getState().sessions
    const data = new FormData()
    const requestHeader = {
      method: "POST",
      headers: new Headers({
        'Authorization': `Token token=${session.session_key}`,
        'Content-Type': 'application/json'
      }),
      body: data.append("json", JSON.stringify({"peep": {"user_id": session.user_id, ...peepData}))
    }
    return fetch("https://chitter-backend-api.herokuapp.com/peeps", requestHeader).then(response => {
      return response.json()
    }).then(json => {
      dispatch(addPeep(json))
    }).catch(e => console.log(e))
  }
}

export const removePeep = ({id} = {}) => ({
  type: 'REMOVE_PEEP',
  id
})

export const setPeeps = (peeps) => ({
  type: 'SET_PEEPS',
  peeps
})

export const startSetPeeps = () => {
  return (dispatch) => {
    return fetch("https://chitter-backend-api.herokuapp.com/peeps").then(response => {
      return response.json()
    }).then(json => {
      dispatch(setPeeps(json))
    }).catch((e) => console.log(e))
  }
}
