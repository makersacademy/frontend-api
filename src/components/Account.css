.ui.card {
  margin: auto !important;
  width: 100% !important;
  height: 100% !important;
}

div.signout.extra.content a {
  color: #4183c4 !important;
}