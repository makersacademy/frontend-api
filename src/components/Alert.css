.ui.message {
  width: 100% !important;
  position: absolute !important;
  display: none;
  bottom: 0;
  left: 0;
  opacity: 0;
  transition: all 2s ease-in-out !important;
}

