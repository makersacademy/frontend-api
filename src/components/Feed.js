import React from 'react'
import { connect } from 'react-redux'
import Peep from './Peep'

export const Feed = (props) => {
  let insert;
  
  console.log(props.peeps)
  if (props.peeps === undefined || props.peeps.length == 0) {
    insert = <p>No peeps yet!</p>
  } else {
    insert = props.peeps.map((peep, index) => <Peep key={index} peep={peep}/>)
  }
  return (
    <div>
      <h1>Feed</h1>
      <ul>
        {insert}
      </ul>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    peeps: state.peeps
  }
}

export default connect(mapStateToProps)(Feed)
