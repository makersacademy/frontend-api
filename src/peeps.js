$(document).ready(function(){
  $.get('https://chitter-backend-api.herokuapp.com/peeps', function(data) {
    data.forEach(function(element) {
      $(".peeps").append("<li>" + element.body + "</li>");
    });

    
  });

  // $(document).ready(function(){
  //   $("#btn1").click(function(){
  //     $("p").append(" <b>Appended text</b>.");
  //   });
  //
  //   $("#btn2").click(function(){
  //     $("ol").append("<li>Appended item</li>");
  //   });
  // });
  //
});
