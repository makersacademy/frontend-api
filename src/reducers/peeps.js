const peepsReducerDefaultState = [];

export default (state=peepsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_PEEP':
      return [action.peep, ...state]
    case 'REMOVE_PEEP':
      return state.filter(({id}) => id !== action.id)
    // case 'EDIT_PEEP':
    //   return state.map((peep) => peep.id === action.id ? {...peep, ...action.updates} ? peep;
    case 'SET_PEEPS':
      return action.peeps
    case 'LIKE_PEEP':
      // const index = state.findIndex((peep) => peep.id === action.id))
      state[state.findIndex((peep) => peep.id === action.id)].likes.push(action.user)
      return state
    default:
      return state
  }
}
