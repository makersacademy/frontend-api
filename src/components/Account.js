import React from "react";
import './Account.css'

const Account = ({ session, setSession }) => {

  const signOut = () => {
    localStorage.removeItem('user')
    setSession(false)
  }

  return (
    <div className="profile">
      <div className="ui card">
        <div className="image">
          <img src={session.avatar} alt='my beautiful pic'/>
        </div>
        <div className="content">
          <a href="#" className="header">{session.handle}</a>
          <div className="meta">
            <span className="date">Joined in 2019</span>
          </div>
          <div className="description">
            <div>Art Director - Los Angeles</div>
            <div>UCLA '15</div>
          </div>
        </div>
        <div className="extra content">
          <a>
            <i className="user icon"></i>
            22 Friends
          </a>
        </div>
        <div className="signout extra content">
          <a className="blue" href="#" onClick={signOut}>Sign Out</a>
        </div>
    </div>
  </div>
  )
}

export default Account