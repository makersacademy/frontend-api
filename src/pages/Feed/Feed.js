import React from "react";
import ListPeeps from "../../components/ListPeeps/ListPeeps";
import Header from "../../components/Header/Header";
import "./Feed.css";

class Feed extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      feed: [],
    };
  }

  extractPeepBodyFromAPIResultArray(array) {
    return array.map((peep) => {
      return peep.body;
    });
  }

  async getPeeps() {
    console.log("get Peeps called");
    const response = await fetch(
      "https://chitter-backend-api-v2.herokuapp.com/peeps"
    );
    const value = await response.json();
    this.setState({ feed: this.extractPeepBodyFromAPIResultArray(value) });
  }

  handleClick() {
    this.getPeeps();
  }

  render() {
    return (
      <>
        <Header title="Welcome to Chitter" />
        <button
          className="button"
          id="refreshFeed"
          onClick={() => this.handleClick()}
        >
          Refresh feed
        </button>
        <ListPeeps feed={this.state.feed} />
      </>
    );
  }
}

export default Feed;
