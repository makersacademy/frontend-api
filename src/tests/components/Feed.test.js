import React from 'react';
import { shallow } from 'enzyme';
import { Feed } from '../../components/Feed'
import peeps from '../fixtures/peeps'

let wrapper

beforeEach(() => {
  wrapper = shallow(<Feed peeps={peeps}/>)
})

test('should render Feed correctly', () => {
  expect(wrapper).toMatchSnapshot();
})
