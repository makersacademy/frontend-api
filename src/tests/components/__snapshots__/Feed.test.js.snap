// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Feed correctly 1`] = `
<div>
  <h1>
    Feed
  </h1>
  <ul>
    <Peep
      key="0"
      peep={
        Object {
          "body": "test peep one",
        }
      }
    />
    <Peep
      key="1"
      peep={
        Object {
          "body": "test peep two",
        }
      }
    />
    <Peep
      key="2"
      peep={
        Object {
          "body": "test peep three",
        }
      }
    />
  </ul>
</div>
`;
